lib/faraday/request.rb

```diff
@@ -42,6 +42,7 @@ def self.create(request_method)
       end
     end
 
+    remove_method :params=
```

Comment:
🗒️ Since `Request` is defined via `Struct.new`, it already has getter and setter methods for its struct attributes, which include `params` and `params=`. The code that follows this statement redefines the `params=` method, which would normally trigger a "method redefined" warning. To avoid that warning, I am explicitly removing the default Struct method first.

I've applied this pattern to the other method definitions that were triggering warnings.

Reply:
Right, that makes sense now!

Before the switch from inheritance (`Request < Struct(...)`) to assignment (`Request = Struct(...)`), these methods were defined in the superclass, so these definitions were simply overriding the parent implementation.
Now they're technically redefinition, which triggers the Ruby warning.

Great catch!